TACC/Core-Portal
# Task/FP-319 Job History Search

## Overview

The goal for this task was to implement a job search feature in the History page. The common searchbar component previously created was used for this. Currently, a job is searchable by `name`, `archiveSystemDir`, `appId` and `archiveSystemId` 

## Related

* [FP-319](https://jira.tacc.utexas.edu/browse/FP-319)
* [FP-1832](https://jira.tacc.utexas.edu/browse/FP-1832)

## Changes

1- The jobs endpoint was changed to accept an `operation` parameter. The operation can be one of the following values: 
     -  `select` used when a specific jobs information needs to be retrieved. (UUID is passed in the request as well)
     - `listing` used when a list of all the jobs needs to be retrieved
     - `search` used when we need to search jobs based on a query string that is passed

2- Added a `fromDashboard` prop to `JobsView` component. JobsView component now has a searchbar and this component is also used on the main dashboard page. To prevent the searchbar from appearing on the dashboard as well this prop was added

3- Minor changes to the common searchbar component which include adding some default prop values and displaying the `dataType` prop value if a `sectionName` is not sent in

## Testing

1. Go to the History page in the portal
2. Make sure you have a few jobs submitted that you can search on
3. Type a query in the searchbar to search jobs by
4. Ensure search functionality works as expected and that records are properly filtered

## UI

<img width="1914" alt="Screen Shot 2023-01-13 at 5 00 02 PM" src="https://user-images.githubusercontent.com/23081932/212434461-ccca55c5-bac3-4b0c-8468-3e463e95524a.png">

<img width="1914" alt="Screen Shot 2023-01-13 at 5 00 22 PM" src="https://user-images.githubusercontent.com/23081932/212434472-1d676670-853b-4797-a3a0-2cd37f2fb380.png">


## Notes

1- All the functionality should be working as expected. Currently working on fixing some of the broken tests
